src/core/util/clone.ts

```diff
@@ -40,23 +63,47 @@ export const getRoot = <T extends Parentable<T>>(
   return element;
 };
 
-/**
- * Determines the corresponding element within the cloned tree.
- * Works as long as the tree was not structurally modified.
- */
-export const getCorrespondingElement = <
-  T1 extends Parentable<T1>,
-  T2 extends Parentable<T2>
->(
-  element: T1,
-  clonedTree: T2
-): T1 | PathError => {
-  const elementPath = calculatePath(getRoot(element), element);
-  if (isPathError(elementPath)) {
-    return elementPath;
+export const findByUUID = (uiSchema: any, uuid: string): any | UUIDError => {
+  const root = getRoot(uiSchema);
+  const element = doFindByUUID(root, uuid);
+  if (!element) {
+    return {
+      id: 'getByUUIDError',
+      root: root,
+      uuid: uuid,
+    };
   }
-  const clonedRoot = getRoot(clonedTree);
-  return getFromPath(clonedRoot, elementPath);
+  return element;
+};
+
+const doFindByUUID = (root: any, uuid: string): any | UUIDError => {
+  if (root && root.uuid === uuid) {
+    return root;
+  }
+  const entries = root instanceof Map ? root.entries() : Object.entries(root);
+  for (const [key, value] of Array.from(entries)) {
+    if (value && value.uuid === uuid) {
+      return value;
+    }
+    // some mappings are 'reversed'
+    if (key && key.uuid === uuid) {
+      return key;
+    }
```

Comment:
These lines could be removed and it should still work as this case is also handled by the other `reversed` handling, right?
```suggestion
```